dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/export/enrollment/EnrollmentOperationParamsMapperTest.java

```diff
@@ -203,6 +207,59 @@ void shouldThrowExceptionWhenOrgUnitNotInScope() {
         exception.getMessage());
   }
 
+  @Test
+  void shouldMapParamsWhenOrgUnitNotInScopeButUserIsSuperuser()
+      throws ForbiddenException, BadRequestException {
+    User superuser = createUser("ALL");
+    when(currentUserService.getCurrentUser()).thenReturn(superuser);
+
+    EnrollmentOperationParams operationParams =
+        EnrollmentOperationParams.builder()
+            .orgUnitUids(Set.of(ORG_UNIT_1_UID))
+            .orgUnitMode(ALL)
+            .build();
+    when(organisationUnitService.isInUserHierarchy(
+            orgUnit1.getUid(), user.getTeiSearchOrganisationUnitsWithFallback()))
+        .thenReturn(false);
+
+    EnrollmentQueryParams queryParams = mapper.map(operationParams);
+    assertContainsOnly(
+        operationParams.getOrgUnitUids(),
+        queryParams.getOrganisationUnits().stream().map(BaseIdentifiableObject::getUid).toList());
+    assertEquals(operationParams.getOrgUnitMode(), queryParams.getOrganisationUnitMode());
+  }
+
+  @Test
+  void shouldFailWhenOrgUnitNotInScopeAndUserHasSearchInAllAuthority() {
```

Comment:
I am confused. What is `SearchInAll` authority adding to the user capabilities?

Reply:
The way I understood it, `SearchInAll` allows the user to run requests with the org unit mode `ALL`, but can't skip user scope validations like a superuser can. That means a user with this authority still needs to have the requested org unit in its search scope in order to see results in the response.

Reply:
I know, for a user who has this authority, I think `ALL`, `ACCESSIBLE` and `DESCENDANTS` in practice are the same thing.